### PR TITLE
IBM Watson speech service implementation

### PR DESCRIPTION
**Pull request by christianahmed**
Implemented a new speech service using IBM Watson for speech transcription and text synthesis.
Speaker diarization is now possible.
Removed the Speech to text flutter plugin.

### CODE REVIEW

**Review by chauntika — APPROVED** [review]
Okay, it worked for me. It's recognizing my voice and continues to listen until I want it to stop or I take a long break. We just have to work on the confidence level, but it is working so far for me on Android.

File: memory_enhancer_app/android/app/src/main/res/values/credentials.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<resources>
+    <!--IBM Speech to Text Service Credentials -->
+    <string name="speech_text_iam_apikey">b1rIMw3vSUe86N7BK7efz6cDorY-g1qTxUPkC2dmmJ6J</string>
+    <string name="speech_text_url">https://api.us-east.speech-to-text.watson.cloud.ibm.com/instances/e8365b62-16bf-48e8-8bd8-cc9b0b556e58</string>
```
Comment (ghost) [inline memory_enhancer_app/android/app/src/main/res/values/credentials.xml:4]: Unless this has been discussed previously and I'm not aware of it, credentials should not be committed. You can use the .gitignore file for this

**Review by ghost — DISMISSED** [review]
Please remove credentials from commit. See https://git-scm.com/docs/gitignore

**Review by ghost — CHANGES_REQUESTED** [review]
Please remove credentials from commit. See https://git-scm.com/docs/gitignore

**Comment by christianahmed** [comment]
> Please remove credentials from commit. See https://git-scm.com/docs/gitignore

@jersoe, the credentials file is used to store the IBM Watson services URL and API key.  I did not want to hardcode them in the code since we will change them many time during this development and each team member will set up their own IBM cloud account; thus having their own credentials.

**Review by ghost — APPROVED** [review]

**Comment by ghost** [comment]
After discussion DevSecOps I've approved this. But as a side note I do want to warn against this.

**Comment by christianahmed** [comment]
> After discussion DevSecOps I've approved this. But as a side note I do want to warn against this.

Thank you Jeroen.  
So that I know how to resolve this later, what is the issue or concern or constraint with saving credentials as such?  That may help me refactor it properly.

**Pull request merged**